plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -535,14 +536,27 @@ class AzBatchService implements Closeable {
         }
     }
 
-    protected VirtualMachineConfiguration poolVmConfig(AzPoolOpts opts) {
+    protected VirtualMachineConfiguration poolVmConfig(String poolId, AzPoolOpts opts) {
```

Comment:
I would avoid adding the `poolId` only for reporting in the log/error message. I think the relevant info is the registry name

Reply:
removed